### PR TITLE
Add Compilation DeviceLab test for Windows

### PR DESCRIPTION
**Pull request by yaakovschectman**
Replaces "Unsupported operation" exceptions in `perf_tests.dart` for Windows OS with a call to the flutter build command for Windows, so that Windows DeviceLab tests can be run. Also adds a compilation test task for Windows, `hello_world_windows__compile`, and a target to `.ci.yaml` to test it. In order to run the newly added task, I also needed to create a windows target build for the `hello_world` example project.

As I do not believe there is any way to test the `.ci.yaml` targets locally, please look over my configuration if you are familiar with these targets to corroborate sure it is correct.

Addresses issue #70027 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: TESTOWNERS [hunk 1]
```diff
@@ -218,6 +218,7 @@
 /dev/devicelab/bin/tasks/run_release_test_macos.dart @cbracken @flutter/tool
 /dev/devicelab/bin/tasks/entrypoint_dart_registrant.dart @aaclarke @flutter/plugin
 /dev/devicelab/bin/tasks/windows_home_scroll_perf__timeline_summary.dart @jonahwilliams @flutter/engine
+/dev/devicelab/bin/tasks/hello_world_windows__compile.dart @schectman @flutter/tool
```
Comment (loic-sharma) [inline TESTOWNERS:221]: Should this be owned by the `@flutter/desktop` team?

**Review by keyonghan — COMMENTED** [review]
ci yaml LGTM.

File: dev/devicelab/lib/tasks/perf_tests.dart [hunk 2]
```diff
@@ -1446,7 +1446,19 @@ class CompileTest {
...
+        options.add('--tree-shake-icons');
+        options.add('--split-debug-info=infos/');
+        watch.start();
+        await flutter('build', options: options);
+        watch.stop();
+        final String basename = cwd.split('/').last;
+        final String exePath = path.join(cwd, 'build', 'windows', 'runner', 'release', '$basename.exe');
+        final File exe = file(exePath);
+        releaseSizeInBytes = exe.lengthSync();
+        // I am unsure if there is any "package" for a windows build, so [reportPackageContentSizes] is irrelevant
```
Comment (cbracken) [inline dev/devicelab/lib/tasks/perf_tests.dart:1460]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-checking-in-comments-that-ask-questions

File: dev/devicelab/lib/tasks/perf_tests.dart [hunk 3]
```diff
@@ -1446,7 +1446,19 @@ class CompileTest {
...
       case DeviceOperatingSystem.windows:
-        throw Exception('Unsupported option for Windows devices');
+        unawaited(stderr.flush());
+        options.insert(0, 'windows');
+        options.add('--tree-shake-icons');
+        options.add('--split-debug-info=infos/');
+        watch.start();
+        await flutter('build', options: options);
+        watch.stop();
+        final String basename = cwd.split('/').last;
```
Comment (cbracken) [inline dev/devicelab/lib/tasks/perf_tests.dart:1456]: Use `path.basename()` instead.

https://pub.dev/documentation/path/latest/path/basename.html

**Review by cbracken — APPROVED** [review]
lgtm modulo comments.

File: TESTOWNERS (hunk 1 above)
Comment (yaakovschectman) [reply TESTOWNERS]: Yes, probably. This has been updated.

**Pull request merged**